pkg/sql/schemachanger/scplan/internal/rules/current/dep_drop_column.go

```diff
@@ -97,4 +97,33 @@ func init() {
 		},
 	)
 
+	// Column constraint transitions to absent right before column transitions
+	// to DELETE_ONLY.
+	// This strengthens and supersedes the above "dependents removed before column"
+	// rule.
+	//
+	// The reason that we don't want column constraint transitions to ABSENT
+	// "too early" (i.e. when column transitions to WRITE_ONLY) is that the
+	// constraint would no longer be enforced while the column is still writable.
+	// This can cause concurrent WRITE to mistakenly write rows into the table
+	// that violates the constraint.
+	//
+	// The reason that we don't want column constraint transition to ABSENT
+	// "too late" (i.e. when column transitions to ABSENT) is that the
+	// constraint would still be visible and enforced while the column is not writable.
+	// This can cause concurrent WRITE to see and attempt to uphold the constraint,
+	// but it wouldn't be able to find the column.
```

Comment:
Thanks for the explanation, I get why it needs to be in the same stage. Any reason why it's the not-null before the column and not the other way round? Not that it matters, just curious.